models/autotetraploid.slim

```diff
@@ -0,0 +1,130 @@
+// Peter Ralph, 27 June 2019: autotetraploidy
+
+/*
+	Tetraploidy:
+	
+	This is a simple model of an autotetraploid.  To do this, we have to 
+	(a) give each individual *four* genomes
+	(b) let crossing-over happen between any of the possible pairings
+	(c) compute fitness with all four genomes
+	
+	(a) Four genomes: the "real" individuals are in p1. Each individual has their other two genomes stashed
+	away in an individual in p2; the individuals in p2 don't do anything on their own. To identify them,
+	each individual in p1 has a "tag" that records the index of their pair in p2, so that
+	  p2.individuals[ind.tag].genomes
+	gets you the other two genomes for individual "ind" in p1. After an individual dies, we clean up their
+	paired individual by setting their fitness to 0.0 in the following generation. So, there's always more
+	individuals in p2 than in p1; the difference is individuals in p2 whose pair in p1 has just died.
+	
+	(b) For crossing-over, we pick an ordering of the mates' four genomes, then do crossing-over by picking
+	a Poisson number of uniform recombination points on each of the four pairs, with an extra crossing-over
+	at 0.0 to randomize which genome we start with. If we want non-uniform recombination, we'd have to get
+	these recombinations some other way.
+	
+	(c) Every new mutation has a property, "h", that is a vector of length 5, always starting with 0 and ending
+	with 1, that has randomly chosen dominance coefficients. Then, the contribution to fitness of having
+	k copies of a mutation is h[k] * s, where recall that: indexing is zero-based; fitness is multiplicative; and
+	since this is a nonWF model, "fitness" means probability of survival until the next generation.
+
+*/
+
+initialize() {
+	initializeSLiMModelType("nonWF");
+		
+	defineConstant("K", 100);
+	defineConstant("L", 1e6); // total bp length
+
+	
+	initializeMutationType("m1", 0.5, "e", 0.01);
+	m1.convertToSubstitution = T;
+	
+	initializeGenomicElementType("g1", m1, 1.0);
+	initializeGenomicElement(g1, 0, L - 1);
+	initializeMutationRate(1e-7);
+	initializeRecombinationRate(1e-8);
+	defineConstant("totalLength", L * 1e-8); // total genetic length
+}
+
+reproduction(p1) {
+	mate = p1.sampleIndividuals(1);
+	ma_chroms = c(individual.genomes, p2.individuals[individual.tag].genomes);
+	pa_chroms = c(mate.genomes, p2.individuals[mate.tag].genomes);
+	pairing = sample(seqLen(4), 4);
+	// including 0.0 starts with the second chromosome instead of the first
+	breakpoints1 = asInteger(c(runif(1) < 0.5 ? 0 else NULL, floor(L * runif(rpois(1, totalLength)))));
```

Comment:
I haven't tested! But, it did not give errors...

Reply:
Hmm.  Pondering it, I don't see any particular reason why it shouldn't work.  But since it is a position that is not between two valid base positions, it is perhaps pushing the boundaries.  If it doesn't present a problem, then it's probably fine.  ;->